app/code/community/Bolt/Boltpay/Model/Admin/ExtraConfig.php

```diff
@@ -181,16 +181,16 @@ public function getExtraConfig($configName, $filterParameters = array() ) {
             : $rawValue;
     }
 
-
     /**
      * Normalizes JSON by stripping new lines from the given string and returning null in the case of only white space.
      * New line characters are added by the text area and this breaks JSON decoding
      *
-     * @param $string   The string to be stripped of newlines
+     * @param string $string    The string to be stripped of newlines
      *
-     * @return string|null    The string stripped of newline characters or null on error
+     * @return string|null      The string stripped of newline characters or null on error
      */
-    private function normalizeJSON($string) {
-        return trim(preg_replace( '/(\r\n)|\n|\r/', '', $string )) ?: array();
+    private function normalizeJSON($string)
+    {
+        return trim(preg_replace( '/(\r\n)|\n|\r/', '', $string )) ?: '';
```

Comment:
This change to returning an array is a very important and very intentional change that was made in version 1.4.0.  Please review Ning's post that is pinned to the top of this thread regarding errors thrown by json_decode on empty stings in PHP 7.0.0

https://app.asana.com/0/inbox/544708310157128/1110257233649801/1110933955640367

Reply:
In one merchant in php5.6 we have a problem with this warning

Reply:
@YevhenShBolt 

I just test your PR on my dev site, the error `Invalid JSON for Bolt Extra Options. Syntax error` is back again, and the PHP version is v7.1.27. Then I suggest to change the empty string back to an empty array.

![image](https://user-images.githubusercontent.com/3198527/55889811-3ddbd900-5be4-11e9-8a7d-1641d457b226.png)